core/src/main/java/org/apache/carbondata/core/scan/expression/conditional/ImplicitExpression.java

```diff
@@ -60,12 +61,16 @@ public ImplicitExpression(Map<String, Set<Integer>> blockIdToBlockletIdMapping)
   private void addBlockEntry(String blockletPath) {
     String blockId =
         blockletPath.substring(0, blockletPath.lastIndexOf(CarbonCommonConstants.FILE_SEPARATOR));
+    // Check if blockletPath contains old tuple id format, and convert it to compatible format.
+    if (blockId.contains("batchno")) {
+      blockId = CarbonTablePath.getShortBlockId(blockId).substring(blockletPath.indexOf('/') + 1);
+    }
     Set<Integer> blockletIds = blockIdToBlockletIdMapping.get(blockId);
     if (null == blockletIds) {
       blockletIds = new HashSet<>();
       blockIdToBlockletIdMapping.put(blockId, blockletIds);
     }
-    blockletIds.add(Integer.parseInt(blockletPath.substring(blockId.length() + 1)));
+    blockletIds.add(Integer.parseInt(blockletPath.substring(blockletPath.lastIndexOf('/') + 1)));
```

Comment:
use File.seperator instead of '/'

Reply:
ok. Changed it to File.seperator instead of '/' in all 3 places.